KVM seems to be broken

**Describe the bug**
When using kvm booting stops with
~~~
KVM: KVM_SET_USER_MEMORY_REGION: File exists
~~~

**To Reproduce**
~~~
$ test/test_dos.py --require-attr=cputest PPDOSGITTestCase
Test PP-DOS-GIT  CPU native vm86 + native DPMI (i386 only)                       ... skipped 'native vm86() only on 32bit x86'
Test PP-DOS-GIT  CPU JIT vm86 + native DPMI                                      ... ok (  2.32s)
Test PP-DOS-GIT  CPU JIT vm86 + JIT DPMI                                         ... ok (  1.98s)
Test PP-DOS-GIT  CPU JIT vm86 + KVM DPMI                                         ... FAIL
Test PP-DOS-GIT  CPU KVM vm86 + KVM DPMI                                         ... FAIL
Test PP-DOS-GIT  CPU KVM vm86 + JIT DPMI                                         ... ok (  1.96s)
Test PP-DOS-GIT  CPU KVM vm86 + native DPMI                                      ... ok (  2.25s)
Test PP-DOS-GIT  CPU KVM vm86 + simulated DPMI                                   ... ok (  9.76s)
Test PP-DOS-GIT  CPU simulated vm86 + simulated DPMI                             ... ok ( 10.12s)
Test PP-DOS-GIT  CPU simulated vm86 + KVM DPMI                                   ... FAIL
Test PP-DOS-GIT  CPU simulated vm86 + native DPMI                                ... ok (  2.39s)
Test PP-DOS-GIT  CPU Trap Flag emulated                                          ... FAIL
Test PP-DOS-GIT  CPU Trap Flag KVM                                               ... FAIL

======================================================================
FAIL: Test PP-DOS-GIT  CPU JIT vm86 + KVM DPMI                                        
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/clients/common/dosemu2.git/test/test_dos.py", line 4778, in test_cpu_jitkvm
    self._test_cpu("emulated", "kvm", 0)
  File "/clients/common/dosemu2.git/test/test_dos.py", line 4759, in _test_cpu
    self.fail("DOS output file not found")
AssertionError: DOS output file not found
Further info in file '/clients/common/dosemu2.git/test_dos.PPDOSGITTestCase.test_cpu_jitkvm.log'
Further info in file '/clients/common/dosemu2.git/test_dos.PPDOSGITTestCase.test_cpu_jitkvm.xpt'

<snip>
~~~

**Attach program/game binaries or provide an URL**
None

**Attach the log**
[t1.zip](https://github.com/dosemu2/dosemu2/files/10687300/t1.zip)

**A regression?**
Yes

**Additional info**
None

Here's the logs with the patch applied.
~~~
$ test/test_dos.py PPDOSGITTestCase.test_cpu_jitkvm
Test PP-DOS-GIT  CPU JIT vm86 + KVM DPMI                                         ... FAIL

======================================================================
FAIL: Test PP-DOS-GIT  CPU JIT vm86 + KVM DPMI                                        
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/clients/common/dosemu2.git/test/test_dos.py", line 4776, in test_cpu_jitkvm
    self._test_cpu("emulated", "kvm", 0)
  File "/clients/common/dosemu2.git/test/test_dos.py", line 4757, in _test_cpu
    self.fail("DOS output file not found")
AssertionError: DOS output file not found
Further info in file '/clients/common/dosemu2.git/test_dos.PPDOSGITTestCase.test_cpu_jitkvm.log'
Further info in file '/clients/common/dosemu2.git/test_dos.PPDOSGITTestCase.test_cpu_jitkvm.xpt'

----------------------------------------------------------------------
Ran 1 test in 1.611s

FAILED (failures=1)
~~~

[t2.zip](https://github.com/dosemu2/dosemu2/files/10696129/t2.zip)

--- a/src/base/emu-i386/kvm.c
+++ b/src/base/emu-i386/kvm.c
@@ -150,7 +150,7 @@ static struct kvm_sregs sregs;
 #define MAXSLOT 400
 static struct kvm_userspace_memory_region maps[MAXSLOT];
 
-static int init_kvm_vcpu(void);
+static int init_kvm_vcpu(dosaddr_t monitor_dosaddr);
 
 static void set_idt_default(dosaddr_t mon, int i)
 {
@@ -228,7 +228,7 @@ void init_kvm_monitor(dosaddr_t monitor_dosaddr)
   /* trap all I/O instructions with GPF */
   memset(monitor->io_bitmap, 0xff, TSS_IOPB_SIZE+1);
 
-  if (!init_kvm_vcpu()) {
+  if (!init_kvm_vcpu(monitor_dosaddr)) {
     leavedos(99);
     return;
   }
@@ -336,7 +336,7 @@ void init_kvm_monitor(dosaddr_t monitor_dosaddr)
 }
 
 /* Initialize KVM and memory mappings */
-static int init_kvm_vcpu(void)
+static int init_kvm_vcpu(dosaddr_t monitor_dosaddr)
 {
   int ret, mmap_size;
 
@@ -346,13 +346,13 @@ static int init_kvm_vcpu(void)
      the kernel needs to emulate that using V86 mode, as is necessary
      on Nehalem and earlier Intel CPUs */
   ret = ioctl(vmfd, KVM_SET_TSS_ADDR,
-	      sregs.tr.base + offsetof(struct monitor, kvm_tss));
+	      monitor_dosaddr + offsetof(struct monitor, kvm_tss));
   if (ret == -1) {
     perror("KVM: KVM_SET_TSS_ADDR\n");
     return 0;
   }
 
-  uint64_t addr = sregs.tr.base + offsetof(struct monitor, kvm_identity_map);
+  uint64_t addr = monitor_dosaddr + offsetof(struct monitor, kvm_identity_map);
   ret = ioctl(vmfd, KVM_SET_IDENTITY_MAP_ADDR, &addr);
   if (ret == -1) {
     perror("KVM: KVM_SET_IDENTITY_MAP_ADDR\n");

